src/LIBSVM.jl

```diff
@@ -223,6 +228,36 @@ function check_train_input(X, y, kernel)
     end
 end
 
+function data2gram(kernel_function, X)
+    ntrain = size(X, 2)
+    gram = Array{Float64}(undef, ntrain, ntrain)
+    for i=1:ntrain
+        for j in 1:i
+            if i==j
+                gram[i,i] = kernel_function(X[:,i], X[:,i])
+            else
+                gram[i, j] = gram[j, i] = kernel_function(X[:, i], X[:, j])
+            end
+        end
+    end
+    return gram
+end
+
+
+function data2gram(kernel, T, SVs::SupportVectors, ntrain::Int)
+    npredict = size(T, 2)
+    gram = zeros(ntrain, npredict)
+
+    for (i, idx) in enumerate(SVs.indices)
+        for j=1:npredict
+            x = SVs.X[:, i]
+            gram[idx,j] = kernel(x, T[:,j])
+        end
+    end
```

Comment:
Can we assume that user defined `kernel` function doesn't modify the input?
```suggestion
    for (i, idx) in enumerate(SVs.indices)
        x = @view SVs.X[:, i]
        for j ∈ 1:npredict
            gram[idx, j] = kernel(x, @view(T[:, j]))
        end
    end
```

Reply:
With `@view` you make it modifiable, right? I cannot imagine a usecase where one would like the kernel to modify it's inputs, the kernel should be a similarity measure and as such a pure function. Otherwise you could run into all kinds of problems like the order of evaluation being important, the kernel evaluating to different values for a datapoint both present in the training and testing dataset, etc.

Reply:
yes, `@view` doesn't copy the data to a new array, and eliminate the memory allocation which is time consuming.

So, I think using `@view` here is reasonable.

Reply:
Ah I think I might've misunderstood your intention, I was thinking you were enabling modification intentionally.

I think it is very unlikely that someone modifies the input, but possible. I don't know how large the performance benefit is but if you think it's worth it, that's fine with me. Maybe we just state that the `kernel` should not modify the input in the README/documentation?

Reply:
> Maybe we just state that the kernel should not modify the input in the README/documentation?

ah, it's fine for me.